src/apps/common/submodules/numberSelector/numberSelector.js

```diff
@@ -228,6 +233,19 @@ define(function(require) {
 							okCallback: addNumberCallback
 						});
 						break;
+					case 'assigned':
+						monster.pub('common.monsterListing.render', {
+							dataList: _.chain(args.numbers.phoneNumbers)
```

Comment:
From looking at this [API request](https://github.com/2600hz/monster-ui/blob/master/src/apps/common/submodules/numberSelector/numberSelector.js#L123), the result [looks to return](https://github.com/2600hz/monster-ui/blob/master/src/js/lib/jquery.kazoosdk.js#L319) all active phone numbers in the system, when here we are only interested in phone numbers assigned to an entity.

This state can be inferred by checking whether `used_by` is set for each phone number returned by our API request. Once that's done, don't forget to update `phoneNumbers` to a more meaningful name, such as `assignedNumbers` or something similar, to clarify what this property holds.